File: Android V1.6/app/src/main/java/com/arcinteractive/spaces/data/model/SpaceList.kt
package com.arcinteractive.spaces.data.model

import java.util.Date
import java.util.UUID

data class SpaceListSection(
    val id: String = UUID.randomUUID().toString(),
    val title: String,
    val order: Int
)

data class SpaceList(
    val id: String,
    val spaceId: String,
    val title: String,
    val sections: List<SpaceListSection>,
    val links: List<SpaceLinkAttachment>,
    val createdBy: String,
    val createdAt: Date,
    val updatedAt: Date
)

data class SpaceListItemAttachment(
    val id: String,
    val name: String,
    val mimeType: String,
    val storagePath: String,
    val nonce: String,
    val isMedia: Boolean
)

data class SpaceListItem(
    val id: String,
    val listId: String,
    val title: String,
    val notes: String,
    val isCompleted: Boolean,
    val assignedMemberIds: Set<String>,
    val dueDate: Date?,
    val sectionId: String?,
    val order: Int,
    val attachments: List<SpaceListItemAttachment>,
    val links: List<SpaceLinkAttachment>,
    val createdBy: String,
    val createdAt: Date,
    val updatedAt: Date
)

File: iOS/Spaces V1.5/Models/SpaceList.swift
import Foundation

struct SpaceListSection: Codable, Identifiable, Hashable {
    var id: String = UUID().uuidString
    var title: String
    var order: Int
}

struct SpaceList: Identifiable, Hashable {
    let id: String
    let spaceID: String
    var title: String
    var sections: [SpaceListSection]
    var links: [SpaceLinkAttachment]
    let createdBy: String
    let createdAt: Date
    var updatedAt: Date
}

struct SpaceListItemAttachment: Codable, Identifiable, Hashable {
    let id: String
    let name: String
    let mimeType: String
    let storagePath: String
    let nonce: String
    let isMedia: Bool
}

struct SpaceListItem: Identifiable, Hashable {
    let id: String
    let listID: String
    var title: String
    var notes: String
    var isCompleted: Bool
    var assignedMemberIDs: Set<String>
    var dueDate: Date?
    var sectionID: String?
    var order: Int
    var attachments: [SpaceListItemAttachment]
    var links: [SpaceLinkAttachment]
    let createdBy: String
    let createdAt: Date
    var updatedAt: Date
}

struct SpaceListPayload: Codable {
    let title: String
    let sections: [SpaceListSection]
    let links: [SpaceLinkAttachment]
}

struct SpaceListItemPayload: Codable {
    let title: String
    let notes: String
    let assignedMemberIDs: [String]
    let attachments: [SpaceListItemAttachment]
    let links: [SpaceLinkAttachment]?
}
